### PR TITLE
fix(oci): Use correct name referencing when tagging

### PR DESCRIPTION
**Pull request by craciunoiuc**
### Prerequisite checklist

  - [x] Read the [contribution guidelines](https://unikraft.org/docs/contributing/) regarding submitting new changes to the project;
  - [x] Tested your changes against relevant architectures and platforms;
  - [x] Ran `make fmt` on your commit series before opening this PR;
  - [ ] Updated relevant documentation.

### CODE REVIEW

File: oci/pack.go [hunk 1]
```diff
@@ -38,7 +38,7 @@ type ociPackage struct {
 	image  *Image
 
 	// Embedded attributes which represent target.Target
-	name    string
+	// name    string
```
Comment (antoineco) [inline oci/pack.go]: If a name is no longer needed, let's remove the field.

File: oci/pack.go [hunk 2]
```diff
@@ -389,7 +387,7 @@ func (ocipack *ociPackage) Pull(ctx context.Context, opts ...pack.PullOption) er
 		return err
 	}
 
-	fullref := ocipack.Name()
+	fullref := fmt.Sprintf("%s:%s", ocipack.Name(), ocipack.Version())
```
Comment (antoineco) [inline oci/pack.go]: nit: I see that this format string is used in two places. We might as well create a very small function to encapsulate this format to ensure that all  functions use the same.

**Review by antoineco — APPROVED** [review]
Approving with a non-blocking comment.

File: oci/pack.go (hunk 2 above)
Comment (nderjung) [reply oci/pack.go]: I looked internally into [the `name` package](github.com/google/go-containerregistry/pkg/name) to see if we could re-use anything there but there was nothing from a first glance.  I've just done a fixup and will push the change now.

**Pull request merged**